src/web/client/dal/remoteFetchProvider.ts

```diff
@@ -364,6 +483,70 @@ async function getMappingEntityContent(
     return result.value ?? Constants.NO_CONTENT;
 }
 
+export async function  preprocessData(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    data: any,
+    entityType: string
+) {
+   try
+   {
+       const schema = WebExtensionContext.urlParametersMap
+               .get(schemaKey.SCHEMA_VERSION)
+               ?.toLowerCase() as string;
+               
+       if (entityType === schemaEntityName.ADVANCEDFORMS && schema.toLowerCase() ===
+           portal_schema_V2.entities.dataSourceProperties.schema)
+       {
+            entityType = schemaEntityName.ADVANCEDFORMSTEPS;
+            const dataverseOrgUrl = WebExtensionContext.urlParametersMap.get(
+                Constants.queryParameters.ORG_URL
+            ) as string;
+            const entityDetails = getEntity(entityType);        
+            const fetchedFileId = entityDetails?.get(schemaEntityKey.FILE_ID_FIELD);
+            const formsData = await fetchFromDataverseAndCreateFiles(entityType, getCustomRequestURL(dataverseOrgUrl, entityType));
+            const attributePath: IAttributePath = getAttributePath(
+                entityAttributeNeedMapping.webformsteps
+            );
+
+            const advancedFormStepData = new Map();
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            formsData.forEach((dataItem: any) => {                
+                const entityId = fetchedFileId ? dataItem[fetchedFileId] : null;
+                if (!entityId) {
+                    throw new Error(ERRORS.FILE_ID_EMPTY);
+                }
+                advancedFormStepData.set(entityId, dataItem);
+            });
+
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            data.forEach((dataItem: any) => {
+                const webFormSteps = GetFileContent(dataItem, attributePath) as []; 
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                const steps: any[] = [];
```

Comment:
can we define Model of steps. Using any is not a good practice 

<img width="544" alt="image" src="https://github.com/microsoft/powerplatform-vscode/assets/97606074/d89147ef-0be7-406a-9a0b-b27152469de2">

Reply:
yeah - this is one code-debt we have to pay at some point - we need to introduce model for all entity codes - atleast the generic ones- will take this up as a corrective PR - we need to convert older model too

Reply:
Added a task on board for this